src/main/java/tutorlink/storage/StudentStorage.java

```diff
@@ -39,23 +42,26 @@ private Student getStudentFromFileLine(String fileLine, ArrayList<Student> stude
             throws InvalidDataFileLineException {
         String[] stringParts = fileLine.split(READ_DELIMITER);
         try {
-            String matricNumber = stringParts[0];
-            String name = stringParts[1];
-            String gpaString = stringParts[2];
-            double gpa = Double.parseDouble(gpaString);
-            Student newStudent = new Student(matricNumber, name, gpa);
+            String matricNumber = stringParts[0].strip().toUpperCase();
+            String name = stringParts[1].strip();
+            Pattern pattern = Pattern.compile(Commons.MATRIC_NUMBER_REGEX);
+            Matcher matcher = pattern.matcher(matricNumber);
```

Comment:
What happens if someone tampers with the storage file and stores a string in the percentageScore section?

Reply:
I removed the percentageScore section from studenlist.txt, instead it is calculated after loading all the storage files.